### PR TITLE
Update doc for CA1021 to mention api surface configurability

### LINKED ISSUE #4615: Update documentation to mention API surface configurability
See https://github.com/dotnet/roslyn-analyzers/pull/3176#discussion_r364393391

This page shall be updated to add the sub-section about API surface compatibility as it is done on https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1000?view=vs-2019#configurability

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0880ab33-6c9a-2541-880b-352e8a68043e
* Version Independent ID: 29e9a293-d06a-5a77-4b7e-cebf29df4c41
* Content: [CA1021: Avoid out parameters - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1021?view=vs-2019)
* Content Source: [docs/code-quality/ca1021.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/code-quality/ca1021.md)
* Product: **visual-studio-windows**
* Technology: **vs-ide-code-analysis**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #4615

This new feature is not yet publicly released. Please sync-up with @mavasani for when to merge this PR.

### CODE REVIEW

**Comment by PRMerger15** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

File: docs/code-quality/ca1021.md [hunk 1]
```diff
@@ -85,5 +87,15 @@ Methods that implement the **Try\<Something>** pattern, such as <xref:System.Int
 ### Code
 [!code-csharp[FxCop.Design.TryPattern#1](../code-quality/codesnippet/CSharp/ca1021-avoid-out-parameters_5.cs)]
 
+## Configurability
```
Comment (mavasani) [inline docs/code-quality/ca1021.md:90]: Please move the `Configurability` section above all the `Example` sections for ordering consistency with existing documentation pages.

**Comment by ktoliver** [comment]
Hi @Evangelink - Could you make the revisions that @mavasani requests? Then we can merge the PR. Thanks.

**Comment by PRMerger6** [comment]
@Evangelink : Thanks for your contribution! The author(s) have been notified to review your proposed change.

**Comment by Mikejo5000** [comment]
#sign-off Looks good.

**Pull request merged**